src/vendor/rlottie/RLottie.svelte

```diff
@@ -1,45 +1,32 @@
 <script lang="ts">
 	import RLottieWasm from "./rlottie-worker";
 
-	let { src }: {
+	let { src, speed }: {
 		src: string;
+		speed: number;
 	} = $props();
 
     let canvasEl = $state<HTMLCanvasElement>();
-
-    class RLottieModule {
-    	canvas: HTMLCanvasElement;
-    	context: CanvasRenderingContext2D;
-    	lottieHandle: any;
-    	totalFrame: number;
-    	curFrame: number;
-
-		constructor() {
-		    this.canvas = canvasEl as HTMLCanvasElement;
-		    this.context = this.canvas.getContext("2d") as CanvasRenderingContext2D;
-		    this.lottieHandle = RLottieWasm;
-		    this.totalFrame = this.lottieHandle.frames();
-		    this.curFrame = 0;
-		}
-
-		render(speed: number) {
-		    if(this.canvas.width == 0 || this.canvas.height == 0) return;
-		    var buffer = this.lottieHandle.render(this.curFrame, this.canvas.width, this.canvas.height);
-		    this.curFrame = Number(this.curFrame + speed);
-		    var result = Uint8ClampedArray.from(buffer);
-		    var imageData = new ImageData(result, this.canvas.width, this.canvas.height);
-		    this.context.putImageData(imageData, 0, 0);
-		}
-	}
-
-	const rm = new RLottieModule();
+    let context = $derived(canvasEl?.getContext("2d"));
+    let lottieHandle: any = $state(RLottieWasm);
+    let totalFrame = $state(lottieHandle.frames());
```

Comment:
Instead of this, use a onMount hook

Reply:
Resolved on https://github.com/baseplate-admin/coreproject-v4-ui/pull/91/commits/5dce8df170046d8a3a6d0cbf60978cfb3a33d749